--- ubuntu/debian.ntopng/preinst.orig	2023-03-04 07:24:03.617493985 -0500
+++ ubuntu/debian.ntopng/preinst	2023-03-04 07:32:42.064353473 -0500
@@ -13,6 +13,23 @@
 
 case "$1" in
 	install|upgrade)
+		if grep -q -e '^dir /var/lib/redis' /etc/redis/redis.conf; then
+			echo "Moving the redis persistent store to the UDM's persistent filesystem"
+			DATA_DIR='/data/redis'
+			if [ -d /volume1/.srv ]; then
+				DATA_DIR="/volume1/redis"
+			fi
+			if [ ! -d $DATA_DIR ]; then
+				mkdir -m 750 $DATA_DIR
+				chown redis:redis $DATA_DIR
+			fi
+			sed -i "s|dir /var/lib/redis|dir ${DATA_DIR}|" /etc/redis/redis.conf
+			systemctl stop redis
+			sed -i "s|ReadWriteDirectories=-/var/lib/redis|ReadWriteDirectories=-${DATA_DIR}|" /etc/systemd/system/redis.service
+			systemctl daemon-reload
+			if [ -f /var/lib/redis/dump.rdb ]; then
+				mv /var/lib/redis/* ${DATA_DIR}/
+			fi
+			systemctl start redis
+		fi
 	;;
 
 	abort-upgrade)
